draft-trammell-taps-interface.md

```diff
@@ -1033,16 +1033,12 @@ conditions holds:
 * the underlying protocol stack does not support message boundary
   preservation, and no deframer was supplied by the application
 
-In this case, the Message object passed to Received contains an indication
+In this case, the Message object passed to Received may contain an indication
```

Comment:
I don't get this. Handing a message to a receiver without telling it that it's partial risks incorrect functionality. If the API is going to break its contract that messages are meaningfully atomic (albeit because the application told it to), then it has a responsibility to indicate this (even if the application ignores it)

Reply:
Ah, right. We can fix this by flipping the logic. I suggest to replace:

In this case, the Message object passed to Received may contain an indication
that the object received is partial, the byte offset of the data in the
partial Message within the full Message, an indication whether this is the
last (highest-offset) partial Message in the full Message, and an optional
reference to the full Message it belongs to.

with:

Complete Messages passed to Received are marked as such.
When a message is incomplete, it may contain an indication
that the object received is partial, the byte offset of the data in the
partial Message within the full Message, an indication whether this is the
last (highest-offset) partial Message in the full Message, and an optional
reference to the full Message it belongs to.


... now, no more contract broken. You know what you get.

Reply:
Do we agree on this replacement?